oryx-app-mllib/src/main/java/com/cloudera/oryx/app/mllib/kmeans/KMeansUpdate.java

```diff
@@ -154,28 +150,30 @@ private PMML kMeansModelToPMML(KMeansModel model, Map<Integer,Long> clusterSizes
     return pmml;
   }
 
-  private ClusteringModel pmmlClusteringModel(KMeansModel model, Map<Integer,Long> clusterSizesMap) {
-    ClusteringModel clusteringModel =
-        new ClusteringModel(AppPMMLUtils.buildMiningSchema(inputSchema),
-            new ComparisonMeasure(ComparisonMeasure.Kind.DISTANCE)
-                .withMeasure(new SquaredEuclidean()),
-            MiningFunctionType.CLUSTERING,
-            ClusteringModel.ModelClass.CENTER_BASED,
-            model.clusterCenters().length)
-            .withNumberOfClusters(model.k());
-
-    if (initializationStrategy.equals(KMeans.K_MEANS_PARALLEL())) {
-      clusteringModel.setAlgorithmName("K-Means||");
-    } else if (initializationStrategy.equals(KMeans.RANDOM())) {
-      clusteringModel.setAlgorithmName("random");
+  private ClusteringModel pmmlClusteringModel(KMeansModel model,
+                                              Map<Integer,Long> clusterSizesMap) {
+    Vector[] clusterCenters = model.clusterCenters();
+    ClusteringModel clusteringModel = new ClusteringModel(
+        AppPMMLUtils.buildMiningSchema(inputSchema),
+        new ComparisonMeasure(ComparisonMeasure.Kind.DISTANCE).withMeasure(new SquaredEuclidean()),
+        MiningFunctionType.CLUSTERING,
+        ClusteringModel.ModelClass.CENTER_BASED,
+        clusterCenters.length);
+
+    for (int i = 0; i < inputSchema.getNumFeatures(); i++) {
+      if (inputSchema.isActive(i)) {
+        FieldName fieldName = FieldName.create(inputSchema.getFeatureNames().get(i));
+        ClusteringField clusteringField =
+            new ClusteringField(fieldName).withCenterField(ClusteringField.CenterField.TRUE);
```

Comment:
I don't think the compare function needs to be overridden, but given my reading of the spec, `centerField` should be true.